### ISSUE TITLE
UseLocalstorage does not preserve uploaded files

### ISSUE THREAD

**Issue by matsn0w** [formkit/formkit#1363]
### Reproduction

https://formkit.link/65bf812002e6e2440628847aaccd6067

### Describe the bug

When saving to local storage, only the file's name is preserved. The actual file itself is lost. This results in a 'false positive' form state in which it looks like the file input is filled, but it actually isn't.

Validation will also pass, but trying to upload this 'file' definitely won't succeed.

I guess this is caused by the way the form data is serialized into JSON.

Can we do something about this?

Thank you in advance and keep up the good work!

### Environment

• OS: macOS Sonoma
• Browser: Google Chrome
• Version: 124.0.6367.209

**Comment by fenilli** [comment]
Yes I would consider that a bug, file types should not keep values from the LocalStorage plugin.

**Comment by obadahalak** [comment]
Any updates??
The issue still exists...

**Comment by justin-schroeder** [comment]
Implemented in PR: https://github.com/formkit/formkit/pull/1738

The localStorage plugin now omits descendant `type="file"` values before saving default cached form data and before loading stale cached data, so file inputs are not rehydrated with name-only metadata that looks valid but cannot upload.

Verification completed:
- Targeted localStorage regression passed.
- Full addons test suite passed.
- Type-only Vitest passed.
- `node scripts/cli.mjs --script build addons` produced successful CJS/ESM addon bundles, then failed during DTS on the pre-existing `@formkit/auto-animate` exports/type resolution issue in `autoAnimatePlugin.ts`.

Security note: this reduces accidental storage of file metadata/objects in localStorage and does not add new storage keys or exposure paths.

**Comment by justin-schroeder** [comment]
This should be fixed in 2.1.0, which was just released. Please upgrade your `@formkit/*` packages to `^2.1.0` and let us know if you're still seeing the issue. Thanks!
